lib/services/message-service.ts

```diff
@@ -22,34 +39,129 @@ import {
+     */
+    protected static extractMetadata(message: string): Metadata {
+        // Retrieve publicity indicators from the environment
+        const visibleArray = JSON.parse(process.env.MESSAGE_CONVERTOR_PUBLIC_INDICATORS);
+        const visible = visibleArray.join('|\\');
+        const hidden = JSON.parse(process.env.MESSAGE_CONVERTOR_PRIVATE_INDICATORS).join('|\\');
+        // Anchored with new line; followed by whitespace.
+        // Captured, the show/hide; brackets to enclose.
+        // Then comes genesis; parens may surround.
+        // The case we ignore; a Regex we form!
+        const findMetadata = new RegExp(`(?:^|\\r|\\n)(?:\\s*)\\[?(${hidden}|${visible})\\]?\\(?(\\w*)\\)?`, 'i');
+        const metadata = message.match(findMetadata);
+        if (metadata) {
+            // The content without the metadata, the word after the emoji, and whether the emoji is in the visible set
+            return {
+                content: message.replace(findMetadata, '').trim(),
+                genesis: metadata[2] || null,
+                hidden: !visibleArray.includes(metadata[1]),
+            };
+        }
+        // Return some default values if there wasn't any metadata
+        return {
+            content: message,
+            genesis: null,
+            hidden: true,
+        };
+    }
+
+    /**
+     * A singleton express instance for all web-hook based message services to share
+     * @type {Express}
+     */
     private static _app: express.Express;
+
+    /**
+     * A boolean flag for if this object has been activated as a listener
+     * @type {boolean}
+     */
     private listening: boolean = false;
+    /**
+     * An object of arrays storing events by trigger and their actions
+     */
     private _eventListeners: { [event: string]: ServiceRegistration[] } = {};
 
     /**
-     * All messenger classes share a single express instance
+     * Create or retrieve the singleton express app
+     * @returns {express.Express}
      */
     protected static get app(): express.Express {
-        // Heroku uses process.env.PORT to indicate which local area port the edge NAT maps down to
-        const port = process.env.MESSAGE_SERVICE_PORT || process.env.PORT;
-        if (!port) {
-            throw new Error('No inbound port specified for express server');
-        }
         if (!MessageService._app) {
+            // Either MESSAGE_SERVICE_PORT from environment or PORT from Heroku environment
+            const port = process.env.MESSAGE_SERVICE_PORT || process.env.PORT;
+            if (!port) {
+                throw new Error('No inbound port specified for express server');
+            }
+            // Create and log an express instance
             MessageService._app = express();
             MessageService._app.use(bodyParser.json());
             MessageService._app.listen(port);
+            MessageService.logger.log(LogLevel.INFO, `---> Started express webserver on port '${port}'`);
```

Comment:
This is just the getting on an app, it doesn't know what it's going to be used for.  It's worth noting that the listeners themselves, when they start listening, also check in.